### ISSUE TITLE
Tuples spread on overloaded function 

### ISSUE THREAD

**Issue by ColCh** [Microsoft/TypeScript#26583]
**TypeScript Version:**  typescript@3.1.0-dev.20180821 

also typescript@stable

**Search Terms:** tuple spread overload

**Code**

```ts
class Docs {
  b(id: string): Promise<{}>;
  b(id: string, options: object): Promise<{}>;
  b(id: string, callback: Function): Promise<{}>;
  b(id: string, options: object, callback: Function): Promise<{}>;
  b(id: string, options?: object, callback?: Function): Promise<{}>;
}
const b = new Docs().b;
type B = typeof b;

const a: B = (a: string) => Promise.resolve({});
const aSpread: B = (...a: [string]) => Promise.resolve({});

a('1234');
```

**Expected behavior:**

`const aSpread` should be same overloaded type as `const a`.

**Actual behavior:** it errors about `a_1`. Though tuple has only 1 member.

**Playground Link:**  http://www.typescriptlang.org/play/

**Related Issues:**  none

**Comment by RyanCavanaugh** [comment]
@ahejlsberg what's going on here?

**Comment by ahejlsberg** [comment]
It does indeed error on the playground, but I can't repro with current master. Clean install of `typescript@next` also fails to reproduce. I'm guessing this was already fixed by #25871. 

@ColCh Can you double check that you're not using an old compiler?

**Comment by ColCh** [comment]
@ahejlsberg indeed, it works with `typesript@next`. I checked with most latest TS version, but ...

... but I was wrong about this button 😂 

![image](https://user-images.githubusercontent.com/599439/44447786-ed325400-a5f1-11e8-9324-4ef7d7c288a0.png)


It seems that this issue is resolved in current master version. I will use it 

Thank you for fast reaction! Closing issue
